hooks/browser_integration.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+class BrowserIntegration(HookBaseClass):
+    def get_cache_key(self, config_uri, project, entity_type):
```

Comment:
i think i might have asked this in a previous PR - how difficult would it be to pass down a full sg data dict instead of project and entity_type? I am thinking about future use cases where we for example want separate menus for shot tasks and asset tasks. in that case, i am guessing we nneed to key the hash off both entity type , config uri and the entity type of the associated entity?